crates/css_ast/src/types/transform/mod.rs

```diff
@@ -0,0 +1,81 @@
+mod matrix;
+mod rotate;
+mod scale;
+mod scale_x;
+mod scale_y;
+mod skew;
+mod skew_x;
+mod skew_y;
+mod translate;
+mod translate_x;
+mod translate_y;
+
+use skew::SkewKind;
+
+use css_lexer::Cursor;
+use css_parse::{Parser, Peek, function_set};
+use csskit_derives::{Parse, ToCursors};
+
+function_set!(TransformFunctionName {
+	Matrix: "matrix",
+	Translate: "translate",
+	TranslateX: "translatex",
+	TranslateY: "translatey",
+	Scale: "scale",
+	ScaleX: "scalex",
+	ScaleY: "scaley",
+	Rotate: "rotate",
+	Skew: "skew",
+	SkewX: "skewx",
+	SkewY: "skewy",
+});
+
+// https://drafts.csswg.org/css-transforms-1/#two-d-transform-functions
+#[derive(Parse, ToCursors, Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize), serde())]
+pub enum TransformFunction {
```

Comment:
This is so convenient, just derive Parse. But can see us _not_ doing that, so individual functions can have simpler impls. And handling the parse steps here. Not sure if individual functions are ever parsed outside of just here, so we wont need the safety that say Matrix::parse_steps has, but checking that the function name is matrix. 

I guess what im trying to ask is, should each fn be able to be parsed on its own? Maybe not, and this enum here should own that logic, and thus simplify all the fns.